core/src/banking_simulation.rs

```diff
@@ -761,6 +761,10 @@ impl BankingSimulator {
             BANKING_TRACE_DIR_DEFAULT_BYTE_LIMIT,
         );
 
+        // Create a partially-dummy ClusterInfo for the banking stage.
+        let cluster_info_for_banking = Arc::new(DummyClusterInfo {
+            id: simulated_leader.into(),
+        });
```

Comment:
iirc cluster info is only needed for forwarding. once I remove forwarding from banking stage (soon after #3820) we can get rid of all this I think.